src/LobbyServer.hs

```diff
@@ -207,9 +210,9 @@ kickPlayerWithGameID remoteGames gameID clientName = do
   liftIO $ CC.modifyMVar_ mVarGamesList $ \lst -> do
     let (h,t) = break ((gameID ==) . fst) lst
     case t of
-      ((_, GameData ps gameName):gt) -> return $ h ++ newGame:gt
+      ((_, gameData):gt) -> return $ h ++ newGame:gt
         where
-          newGame = (gameID, GameData (filter ((clientName /=) . name) ps) gameName)
+          newGame = (gameID, gameData {players = filter ((clientName /=) . name) $ players gameData})
```

Comment:
Code duplication. with 232.
Could be extracted to a named function, say `deletePlayerFrom game` ,some game module or utils or to LobbyTypes for now. Also take a look at the `deleteBy` function on hoogle. It only deletes the first occurance but i see no reason why that would't be enough. Since player names should be unique (they aren't right now but they should be in the future ,right?) and a player should not be able to join a game twice.